File: test/language/templates/cpp/StructLongTemplateArgumentTest.cpp
#include "gtest/gtest.h"
#include "templates/struct_long_template_argument/StructLongTemplateArgument.h"

namespace templates
{
namespace struct_long_template_argument
{

TEST(StructLongTemplateArgumentTest, readWrite)
{
    const TemplatedStruct_ThisIsVeryVeryVeryLongNamedStructure_ThisIsVery_ templ(
            ThisIsVeryVeryVeryLongNamedStructure("StringT"), ThisIsVeryVeryVeryLongNamedStructure("StringU"),
            ThisIsVeryVeryVeryLongNamedStructure("StringV"));
    StructLongTemplateArgument structLongTemplateArgument(templ);

    zserio::BitBuffer bitBuffer = zserio::BitBuffer(1024 * 8);
    zserio::BitStreamWriter writer(bitBuffer);
    structLongTemplateArgument.write(writer);

    zserio::BitStreamReader reader(writer.getWriteBuffer(), writer.getBitPosition(), zserio::BitsTag());
    StructLongTemplateArgument readStructLongTemplateArgument(reader);

    ASSERT_TRUE(structLongTemplateArgument == readStructLongTemplateArgument);
}

} // namespace struct_long_template_argument
} // namespace templates

File: test/language/templates/python/StructLongTemplateArgumentTest.py
import zserio

import Templates

class StructLongTemplateArgumentTest(Templates.TestCase):
    def testReadWrite(self):
        templ = self.api.TemplatedStruct_ThisIsVeryVeryVeryLongNamedStructure_ThisIsVery_(
            self.api.ThisIsVeryVeryVeryLongNamedStructure("StringT"),
            self.api.ThisIsVeryVeryVeryLongNamedStructure("StringU"),
            self.api.ThisIsVeryVeryVeryLongNamedStructure("StringV")
        )
        structLongTemplateArgument = self.api.StructLongTemplateArgument(templ)

        writer = zserio.BitStreamWriter()
        structLongTemplateArgument.write(writer)
        reader = zserio.BitStreamReader(writer.byte_array, writer.bitposition)
        readStructLongTemplateArgument = self.api.StructLongTemplateArgument()
        readStructLongTemplateArgument.read(reader)
        self.assertEqual(structLongTemplateArgument, readStructLongTemplateArgument)
